app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,33 @@
+class SessionsController < ApplicationController
+  before_action :find_user, only: %i[show edit update]
+  def show; end
+  def new
+
+  end
+
+  def index
+    @users = User.all
+  end
+
+  def create
+    user = User.authenticate(params[:username], params[:password])
+    if user
+      session[:user_id] = user.id
+      redirect_to({ action: :index }, notice: 'Login up!')
+    else
+      flash.now.alert = 'Invalid email or password'
+      render :new
+    end
+    end
+
+    def destroy
+      session[:user_id] = nil
+      redirect_to({ action: :index }, notice: 'Logged out!')
+    end
+end
+
+private
```

Comment:
@SemyzSerhiy why this declaration is out of the class definition? Check this point please on your side. I am pretty sure that this is a typo

Reply:
@SemyzSerhiy same for find_user method